rollup/sync_service/bridge_client.go

```diff
@@ -0,0 +1,170 @@
+package sync_service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum"
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rpc"
+)
+
+// BridgeClient is a wrapper around EthClient that adds
+// methods for conveniently collecting L1 messages.
+type BridgeClient struct {
+	client                EthClient
+	confirmations         rpc.BlockNumber
+	l1MessageQueueAddress common.Address
+}
+
+func newBridgeClient(ctx context.Context, l1Client EthClient, l1ChainId uint64, confirmations rpc.BlockNumber, l1MessageQueueAddress *common.Address) (*BridgeClient, error) {
```

Comment:
also change here to be a pointer?
```suggestion
func newBridgeClient(ctx context.Context, l1Client *EthClient, l1ChainId uint64, confirmations rpc.BlockNumber, l1MessageQueueAddress *common.Address) (*BridgeClient, error) {
```